add labels and change colors

Vahlaz opened the pull request:
## Changes

- Added labels for the areas on the cumulative flow diagram
- Changed colors for the areas on the cmf
- Tooltip now shows hovered area bolded 

## Checklist (check all that apply, once you have confirmed they are OK)

- [x] **Styles:**
  1. No global styles.
  2. Root element (App component) should not have any specific styles.
  3. Utilize the MUI (Material-UI) library for styling.
  4. Prefer creating styled components using MUI or custom styling for specific classes/IDs.
  5. Define styles using `sx` prop rather than `style` prop.

- [x] **File Naming and Folder Structure:**
  1. File naming conventions: React components in PascalCase, others in camelCase
  2. Database types and api logic is handled in an `entities` directory.
  3. 'Pages' directory for primary page components handled by the router.
  4. Prefer one React component per file.

- [x] **General Guidelines:**
  1. Limit additional types/interfaces in `.tsx` files to main function props; place other types/interfaces in separate type files.
  2. Avoid unnecessary comments.
  3. Use arrow notation, e.g. `const xxx = () => {}` for defining functions.
  4. No `console.log()` in the code.

CommodoreM reviewed (commented):
The commit does what it's supposed to do and the code is clean. Most of the variable names are very intuitive which is very nice because it makes the changes easy to follow.

One small change request and some observations:

Change request:
line 70: "const dataLength = data?.data.length || 0" If dataLength is set to zero, the graph page is not rendered at all (possibly crashes). The reason why I'm bringing it up, is because of the OR statement which I assume is some sort of protection. It is very possible dataLength is never zero, but in that case the protection should maybe not be there.

some observations:
line 79 and lines: 83 - 86: 
  "const labelYvalues = lastVals.map((val) => {
    sum += val
    return sum - 0.5 * val
  })"

Comments on functions like this may be helpful, because they contain numeric values with no explanation where the values come from. It is easy enough to find out what they do, especially if one tinkers with the values. The project has very few comments so it is definitely a question of preference for the author.

Finally, if there is a lot of data on a board, when the chart is drawn, the labels on right side start to mix up and it becomes harder to read them. This may be a hard problem to avoid and in the future there may be an option to select what is displayed, so it may not ever be an issue.

![labelsMixingUp](https://github.com/user-attachments/assets/88f79cf4-e820-4eab-8bd4-610de73cf93a)

Vahlaz commented:
@CommodoreM The data check is now before getting the length. Added a max length for the label, so column names that are too long are trimmed, also adjusted the minimum tick gap so they cant be as close to eachother.

CommodoreM reviewed (commented):
The loading animation is a nice touch. It is very noticeable in the slow development environment. The changes look good, only one small thing to note:

In certain conditions, rows do not recieve a tick name on the y-axis even if they have a decent number of tickets ( "sprint 4 done" with tickets starting from 2 to 38), picture:

<img width="477" alt="Sprint4DoneNotVisibleOnTheRightSide" src="https://github.com/user-attachments/assets/ff5dc0a3-b0ec-4afe-a09b-f643e941abe1" />

Adding 3 columns to this particular board caused the graph to render a tick name for "sprint 4 done" , picture:

<img width="952" alt="AddingSomeColumnsBreaksTheLogicSprint4HasATickNowToo" src="https://github.com/user-attachments/assets/6281d568-0254-4da4-9207-ff4557bfe915" />

Then, removing two of the added columns results in a graph where row "sprint 4 done" does not have a tick again, even though it had one and it sill has a considerable amount of tickets.

<img width="547" alt="RemovingTwoColumnsMakesTheTickForSprint4Dissapear" src="https://github.com/user-attachments/assets/a72aeeb5-1c4f-418a-802c-e9b6fa4ea310" />

Vahlaz commented:
@CommodoreM Did some fine tuning but probably impossible to get it completely perfect without building custom component and more work

CommodoreM reviewed (approved):
LGTM, the chart is able to support an impressive number of simultaneous columns, good stuff!
